adam/learner/propose_but_verify.py

```diff
@@ -0,0 +1,372 @@
+import logging
+from pathlib import Path
+from random import Random
+from typing import Dict, Generic, List, Mapping, Optional, Set, Tuple
+
+from attr.validators import instance_of, optional
+from immutablecollections import immutabledict
+from vistautils.parameters import Parameters
+
+from adam.language import (
+    LinguisticDescription,
+    LinguisticDescriptionT,
+    TokenSequenceLinguisticDescription,
+)
+from adam.learner import (
+    LanguageLearner,
+    LearningExample,
+    get_largest_matching_pattern,
+    graph_without_learner,
+)
+from adam.ontology.ontology import Ontology
+from adam.ontology.phase1_ontology import GAILA_PHASE_1_ONTOLOGY
+from adam.ontology.phase1_spatial_relations import Region
+from adam.perception import ObjectPerception, PerceptionT, PerceptualRepresentation
+from adam.perception.developmental_primitive_perception import (
+    DevelopmentalPrimitivePerceptionFrame,
+    RgbColorPerception,
+)
+from adam.perception.perception_graph import (
+    DebugCallableType,
+    GraphLogger,
+    PerceptionGraph,
+    PerceptionGraphPattern,
+)
+from adam.utils import networkx_utils
+from attr import Factory, attrib, attrs
+
+
+@attrs()
+class ProposeButVerifyLanguageLearner(
```

Comment:
Can you add a docstring briefly explaining what this learning algorithm does - with a link to the paper that explains the algorithm?

Reply:
Added